Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/CallBAP.hs

```diff
@@ -129,10 +107,20 @@ callOnSelectV2 transporter searchRequest searchTry content = do
       authKey = getHttpManagerKey bppSubscriberId
   bppUri <- buildBppUrl (transporter.id)
   internalEndPointHashMap <- asks (.internalEndPointHashMap)
-  let msgId = searchTry.estimateId
+
+  msgId <- getMsgIdByTxnId searchRequest.transactionId
   context <- ContextV2.buildContextV2 Context.ON_SELECT Context.MOBILITY msgId (Just searchRequest.transactionId) bapId bapUri (Just bppSubscriberId) (Just bppUri) (fromMaybe transporter.city searchRequest.bapCity) (fromMaybe Context.India searchRequest.bapCountry)
   logDebug $ "on_selectV2 request bpp: " <> show content
   void $ withShortRetry $ Beckn.callBecknAPI (Just $ ET.ManagerSelector authKey) Nothing (show Context.ON_SELECT) API.onSelectAPIV2 bapUri internalEndPointHashMap (Spec.OnSelectReq context Nothing (Just content))
+  where
+    getMsgIdByTxnId :: CacheFlow m r => Text -> m Text
+    getMsgIdByTxnId txnId = do
+      Hedis.withCrossAppRedis (Hedis.safeGet $ mkTxnIdKey txnId) >>= \case
```

Comment:
Why are we using `withCrossAppRedis`? this can be driver-app specific right?

Reply:
fixed.